File: Day 3/gcd.java
import java.util.Scanner;

public class gcd {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter first number: ");
        int n1 = sc.nextInt();
        System.out.print("Enter second number: ");
        int n2 = sc.nextInt();
        
        int temp1 = Math.abs(n1);
        int temp2 = Math.abs(n2);
        
        // Euclidean Algorithm
        while (temp2 != 0) {
            int rem = temp1 % temp2;
            temp1 = temp2;
            temp2 = rem;
        }
        
        System.out.println("GCD of " + n1 + " and " + n2 + " is: " + temp1);
        sc.close();
    }
}

File: Day 3/gcd.c
#include <stdio.h>

int main() {
    int n1, n2, temp1, temp2, rem;

    printf("Enter first number: ");
    scanf("%d", &n1);
    printf("Enter second number: ");
    scanf("%d", &n2);

    // Copy original numbers to temporary variables to keep originals safe
    temp1 = n1;
    temp2 = n2;

    // Euclidean Algorithm using a while loop
    while (temp2 != 0) {
        rem = temp1 % temp2;
        temp1 = temp2;
        temp2 = rem;
    }

    // After the loop finishes, temp1 holds the GCD
    printf("GCD of %d and %d is: %d\n", n1, n2, temp1);

    return 0;
}